src/peppr/rotamer_rama.py

```diff
@@ -0,0 +1,1259 @@
+        A dictionary containing the rotamer classification results:
+        {
+            "model_no": int,
+            "resname": str,
+            "resid": int,
+            "observed": dict,
+            "chain_id": str,
+            "rotamer_score_pct": float,
+            "classification": str,
+            "error": str | None
+        }.
+    """
+    if not isinstance(atom_array, struc.AtomArray):
+        raise TypeError("atom_array must be a biotite structure AtomArray")
+    mask = (atom_array.chain_id == chain_id) & (atom_array.res_id == res_id)
+    if not np.any(mask):
+        raise ValueError("Residue not found")
+
+    resname = atom_array.res_name[mask][0].upper()
+    observed = get_residue_chis(atom_array, mask)
+    if not observed:
+        return {
+            "model_no": model_no,
+            "resname": resname,
+            "resid": int(res_id),
+            "chain_id": chain_id,
+            "observed": observed,
+            "rotamer_score_pct": 1.0,  # ALA and GLY have no chi angles
+            "classification": "UNKNOWN",
+            "error": "no chis computable",
+        }
+
+    grid_obj = load_contour_grid_for_residue(
+        grid_dirname="Top8000_rotamer_pct_contour_grids",
+        resname_tag=grid_residue_map.get(resname),
+        grid_data_tag="rota8000",
+        files_to_skip=("rota8000-leu-raw.data",),  # Use rota8000-leu.data instead
+    )
+
+    # Build coords in same order as grid dims
+    coords_list = []
+    for i in range(len(grid_obj["axes"])):
+        coords_list.append(observed.get(f"chi{i + 1}", 0.0))
+
+    pct, wrapped_angles = interp_wrapped(grid_obj, coords_list)
+    observed = {f"chi{i + 1}": wrapped_angles[i] for i in range(len(wrapped_angles))}
+
+    if np.isnan(pct):
+        classification = "UNKNOWN"
+
+    elif pct >= ROTA_ALLOWED_THRESHOLD:
+        classification = "FAVORED"
+    elif pct >= ROTA_OUTLIER_THRESHOLD:
+        classification = "ALLOWED"
+    else:
+        classification = "OUTLIER"
+
+    return {
+        "model_no": model_no,
+        "resname": resname,
+        "resid": int(res_id),
+        "observed": observed,
+        "chain_id": chain_id,
+        "rotamer_score_pct": pct,
+        "classification": classification,
+        "error": None,
+    }
+
+
+def get_residue_phi_psi_omega(atom_array: struc.AtomArray) -> dict:
+    """
+    Get the phi, psi, and omega dihedral angles for a given residue.
+
+    Parameters
+    ----------
+    atom_array : struc.AtomArray
+        The AtomArray containing the structure.
+
+    Returns
+    -------
+    dict[str, np.ndarray]
+        A dictionary containing the phi, psi, and omega angles in degrees.
+    """
+    if atom_array.shape[0] < 4:
+        raise ValueError(
+            "atom_array must contain at least 4 atoms for dihedral angle calculation"
+        )
+    phi, psi, omega = struc.dihedral_backbone(atom_array)
+    phi = np.rad2deg(phi)
+    psi = np.rad2deg(psi)
+    omega = np.rad2deg(omega)
```

Comment:
Calling this before the NaN values are removed will probably raise a warning from NumPy. Hence, we can probably move this after the filtering.